components/tidb_query/src/rpn_expr/impl_cast.rs

```diff
@@ -179,6 +204,140 @@ fn in_union(implicit_args: &[ScalarValue]) -> bool {
     implicit_args.get(0) == Some(&ScalarValue::Int(Some(1)))
 }
 
+// cast any as real, some cast functions reuse `cast_any_as_any`
+//
+// cast_decimal_as_signed_real -> cast_any_as_any<Decimal, Real>
+// cast_time_as_real -> cast_any_as_any<Time, Real>
+// cast_duration_as_real -> cast_any_as_any<Duration, Real>
+// cast_json_as_real -> by cast_any_as_any<Json, Real>
+
+#[rpn_fn]
+#[inline]
+fn cast_signed_int_as_signed_real(val: &Option<Int>) -> Result<Option<Real>> {
+    match val {
+        None => Ok(None),
+        Some(val) => Ok(Real::new(*val as f64).ok()),
+    }
+}
+
+#[rpn_fn(capture = [extra])]
+#[inline]
+fn cast_signed_int_as_unsigned_real(
+    extra: &RpnFnCallExtra,
+    val: &Option<Int>,
+) -> Result<Option<Real>> {
+    match val {
+        None => Ok(None),
+        Some(val) => {
+            if in_union(extra.implicit_args) && *val < 0 {
+                Ok(Some(Real::zero()))
+            } else {
+                // FIXME, TiDB here may has bug, fix this after fix TiDB's
+                Ok(Real::new(*val as u64 as f64).ok())
+            }
+        }
+    }
+}
+
+// because we needn't to consider if uint overflow upper boundary of signed real,
+// so we can merge uint to signed/unsigned real in one function
+#[rpn_fn]
+#[inline]
+fn cast_unsigned_int_as_signed_or_unsigned_real(val: &Option<Int>) -> Result<Option<Real>> {
```

Comment:
Can we shorter the function name? e.g: `cast_uint_as_real`?

Reply:
Because I need to separate it from `cast_signed_int_as_signed_real` and `cast_signed_int_as_unsigned_real`.